samples/apps/copilot-chat-app/webapi/Controllers/DocumentImportController.cs

```diff
@@ -0,0 +1,181 @@
+        }
+
+        this._logger.LogInformation("Importing document {0}", formFile.FileName);
+
+        try
+        {
+            var fileType = this.GetFileType(Path.GetFileName(formFile.FileName));
+            var fileContent = string.Empty;
+            switch (fileType)
+            {
+                case SupportedFileType.TXT:
+                    fileContent = await this.ReadTxtFileAsync(formFile);
+                    break;
+                case SupportedFileType.PDF:
+                    fileContent = await this.ReadPdfFileAsync(formFile);
+                    break;
+                default:
+                    break;
+            }
+            await this.ParseDocumentContentToMemoryAsync(kernel, fileContent, documentImportForm);
+        }
+        catch (Exception ex) when (ex is ArgumentOutOfRangeException)
+        {
+            return this.BadRequest(ex.Message);
+        }
+
+        return this.Ok();
+    }
+
+    /// <summary>
+    /// Get the file type from the file extension.
+    /// </summary>
+    /// <param name="fileName">Name of the file.</param>
+    /// <returns>A SupportedFileType.</returns>
+    /// <exception cref="ArgumentOutOfRangeException"></exception>
+    private SupportedFileType GetFileType(string fileName)
+    {
+        string extension = Path.GetExtension(fileName);
+        return extension switch
+        {
+            ".txt" => SupportedFileType.TXT,
+            ".pdf" => SupportedFileType.PDF,
+            _ => throw new ArgumentOutOfRangeException($"Unsupported file type: {extension}"),
+        };
+    }
+
+    /// <summary>
+    /// Read the content of a text file.
+    /// </summary>
+    /// <param name="file">An IFormFile object.</param>
+    /// <returns>A string of the content of the file.</returns>
+    private async Task<string> ReadTxtFileAsync(IFormFile file)
+    {
+        using var streamReader = new StreamReader(file.OpenReadStream());
+        return await streamReader.ReadToEndAsync();
+    }
+
+    /// <summary>
+    /// Read the content of a PDF file.
+    /// </summary>
+    /// <param name="file">An IFormFile object.</param>
+    /// <returns>A string of the content of the file.</returns>
+    private Task<string> ReadPdfFileAsync(IFormFile file)
+    {
+        throw new ArgumentException("PDF file is not supported yet.");
+    }
+
+    /// <summary>
+    /// Parse the content of the document to memory.
+    /// </summary>
+    /// <param name="kernel">The kernel instance from the service</param>
+    /// <param name="content">The file content read from the uploaded document</param>
+    /// <param name="documentImportForm">The document upload form that contains additional necessary info</param>
+    /// <returns></returns>
+    private async Task ParseDocumentContentToMemoryAsync(Kernel kernel, string content, DocumentImportForm documentImportForm)
+    {
+        var config = this._configuration.GetSection("DocumentImport").Get<DocumentImportConfig>();
```

Comment:
Since chunking/parsing is becoming a conversation piece, could you add a comment block here explaining how we are chunking the document?